sum.go

```diff
@@ -31,6 +41,23 @@ func Sum(data []byte, code int, length int) (Multihash, error) {
 		d, err = sumSHA3(data)
 	case DBL_SHA2_256:
 		d = sumSHA256(sumSHA256(data))
+	case BLAKE2B:
+		switch length {
+		case 32:
+			out := blake2b.Sum256(data)
+			d = out[:]
+		case 48:
+			out := blake2b.Sum384(data)
+			d = out[:]
+		case 64:
+			out := blake2b.Sum512(data)
+			d = out[:]
+		default:
+			return nil, fmt.Errorf("unsupported length for blake2b")
+		}
```

Comment:
> "byte 0x78 means its blake2, then you take the next byte, and thats its length, then the next byte is also its length"

I think you misunderstood. AFAIU, @Kubuxu means that `0xb201 - 0xb240` is a range in the varint hash function code. meaning a bunch of function codes in multicodec. Same for blake2s. this is plain multihash. `<fn-code><len><val>`